source/Politics.cpp

```diff
@@ -226,6 +223,16 @@ string Politics::Fine(PlayerInfo &player, const Government *gov, int scan, const
 		
 		if(!scan || (scan & ShipEvent::SCAN_CARGO))
 		{
+			// Illegal goods can be hidden inside legal goods to avoid detection.
+			const auto contraband = ship->Cargo().IllegalCargoAmount();
+			const auto concealment = ship->Attributes().Get("concealment");
+			const auto legalGoods = ship->Cargo().Used() - contraband;
+
+			const auto netIllegalCargo = contraband - concealment;
+			const auto scaledLegalGoods = legalGoods * (1. + ship->Attributes().Get("scan interference"));
```

Comment:
Not a fan of sprinkling `auto` everywhere, especially with such primitive types. Like come on, some of these are `int`s. It'd be quicker to type `int` than `auto`.